airflow/providers/amazon/provider.yaml

```diff
@@ -52,6 +52,7 @@ dependencies:
   - apache-airflow>=2.3.0
   - apache-airflow-providers-common-sql>=1.3.1
   - boto3>=1.24.0
+  - aiobotocore>=2.1.1
```

Comment:
I think add this as core dependency for amazon-provider could broke some existed functionality.
At least until `aiobotocore` resolve this issue https://github.com/aio-libs/aiobotocore/issues/976 and remove upper bound of [`botocore`](https://github.com/aio-libs/aiobotocore/blob/14b2f5fe0033a1f4bbc456f10633c2aeea74f57a/setup.py#L10).

Because `botocore` is a main package for `boto3` and if we can't upgrade both of them we can add new features which introduced since `botocore` 1.28.0+

Reply:
hmm, how we can avoid this conflict? 
should we keep this optional and move related imports to blocks where it getting used 🤷‍♂️ ?
once we will conclude on this I'll address the other comments

Reply:
@kaxil @uranusjr @ashb  any thoughts on this?

Reply:
Maybe add a comment linking to that issue and a note to change it to a dependency once that is sorted?

Reply:
sure, thanks @ferruzzi @kaxil @phanikumv I'll address the other feeback soon.

Reply:
@ferruzzi I don't think we can move it to core dependency even if it resolve now, there is about 3-5 new version of `botocore` per week and in opposite `aiobotocore` usually [bump botocore dependency](https://github.com/aio-libs/aiobotocore/commits/master/setup.py) once per 3-6 months.

Unfortunetly nothing we could do better until boto-team release next major version of botocore with asyncio support: https://github.com/boto/botocore/issues/458

Reply:
but @Taragolis we can move it out from the core dependency and if someone wants to use the async operator then they will have to install the `aiobotocore`?

Reply:
Yeah we could catch that module not installed and raise appropriate exception, for example

https://github.com/apache/airflow/blob/a7e1cb2fbfc684508f4b832527ae2371f99ad37d/airflow/providers/slack/notifications/slack_notifier.py#L26-L31

But also need somehow resolve additional issues:
1. AFAIK in CI image only core providers dependencies installed (i guess we have some workaround). That mean we have an error in async Amazon providers tests because there is no `aiobotocore`
2. If we install `aiobotocore` in CI image than we also freeze version of `botocore` and `boto3`. As result we could miss if our tests not work with modern version of `boto3` (we have some private method usage in Amazon Provider)

I'm not sure how to resolve this, may be @potiuk could suggest something

Reply:
@potiuk WDYT ^^?

Reply:
Sorry - I totally missed that. Yes we have full support for that:
1) raise AirflowOptionalProviderFeatureException as mentioned above
2) Add "additional-extra" in the amazon provider - same as pandas we already have there: https://github.com/apache/airflow/blob/main/airflow/providers/amazon/provider.yaml#LL342-L597C23

Reply:
> 1. AFAIK in CI image only core providers dependencies installed (i guess we have some workaround). That mean we have an error in async Amazon providers tests because there is no aiobotocore

Yes. It is quite a bummer that  aiobotocore limits botocore that much. Indeed if aiobotocore will be additional extra and will be missing in the image.  We could add it, but as @Taragolis mentions, it would limit our botocore upgrades that happen regularly. A solution to that might be:

1) add exclusion in the tests when iobotocore is installed
2) implement an extra job in CI that installs iobotocore and then runs only the iobotocore tests.

Should be easy to do.

Reply:
I have excluded the test from the regular provider CI job and added a ci. job to run aiobotocore-related tests. can you please re-review it. thanks